### PR TITLE
Add the iModel.js Docs build as yaml

### PR DESCRIPTION
**Pull request by naveedkhan8067**
This pull request replaces a classic build pipeline that builds the itwinjs documentation site artifacts from the various sources within this repository, and some other currently internal repositories.

The reason for the switch is to allow for easier structural build changes to the itwinjs-core repo without creating a mess of conditionals all over the classic release pipeline that makes maintenance difficult.

> Note: Over time we will be moving the other sources of docs into the open (primarily the BIS related schema docs). The timeframe for that is currently to be determined.

The steps in order to get this PR completed:

- [x]  Review the PR to ensure no steps are missing from the current pipeline into this one (@williamkbentley, @BaoTon and @calebmshafer should review)
    - [ ]  Correctly handle the .updated.json file
- [ ]  Determine strategy for updating the release pipeline to handle both this build and the 2.x and 1.x branches.
- [x]  Fix the commits that were committed with a bad email address that cannot sign the CLA (@naveedkhan8067)

### CODE REVIEW

File: common/config/azure-pipelines/templates/docs-build.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+- script: npm install @bentley/bemetalsmith@4.x
+  displayName: Install BeMetalsmith
+  workingDirectory: $(Build.ArtifactStagingDirectory)
+
+- script: "./node_modules/.bin/bmsBuild --strictLinkChecking --topicsMustHaveDesc --source ${{ parameters.stagingDir }} --destination ${{ parameters.outputDir }} --siteTitle iTwin.js"
+  displayName: Run bmsBuild
+  workingDirectory: $(Build.ArtifactStagingDirectory)
+
+# Update the .updated.json file after the bmsBuild
+- task: PublishPipelineArtifact@1
```
Comment (williamkbentley) [inline common/config/azure-pipelines/templates/docs-build.yaml:48]: This only needs to be published from master, when prepping for a release

File: common/config/azure-pipelines/templates/gather-docs.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      artifactName: Bis Docs
+    condition: and(succeeded(), eq('${{ parameters.downloadCurrentBuildArtifacts }}', true))
+
+  - task: CopyFiles@2
+    displayName: Copy Bis Docs to staging
+    inputs:
+      SourceFolder: "$(System.ArtifactsDirectory)/Bis Docs"
+      TargetFolder: ${{ parameters.stagingDir }}/bis/domains/
+
+  # The .updated.json is cached in order to let BeMetalsmith know what needs to be re-built.
```
Comment (williamkbentley) [inline common/config/azure-pipelines/templates/gather-docs.yaml:64]: This comment isn't strictly correct. .updated.json is used for "Last updated" date on each page

File: common/config/azure-pipelines/templates/docs-build.yaml (hunk 1 above)
Comment (naveedkhan8067) [reply common/config/azure-pipelines/templates/docs-build.yaml:48]: @calebmshafer

File: common/config/azure-pipelines/templates/gather-docs.yaml (hunk 2 above)
Comment (naveedkhan8067) [reply common/config/azure-pipelines/templates/gather-docs.yaml:64]: Let me know what you want to write here. I can update it although its not from my changes :)

Comment (naveedkhan8067) [reply common/config/azure-pipelines/templates/gather-docs.yaml]: Done

File: common/config/azure-pipelines/templates/docs-build.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+steps:
+- script: node common/scripts/install-run-rush.js install
+  displayName: rush install
+  workingDirectory: ${{ parameters.workingDir }}
+
+- script: node common/scripts/install-run-rush.js build -v
+  displayName: rush build
+  workingDirectory: ${{ parameters.workingDir }}
+
+- script: node common/scripts/install-run-rush.js docs
```
Comment (williamkbentley) [inline common/config/azure-pipelines/templates/docs-build.yaml:27]: I've also always thought that we should be able to: 
- Parse rush.json
- Make a list of the packages with `shouldPublish=true`
- Pass that list to `rush build` and `rush docs` with `-t` option
 
Should speed up the builds a bit by not building and running Typedoc on undocumented/unpublished packages (test apps and test packages come to mind)

Comment (calebmshafer) [reply common/config/azure-pipelines/templates/docs-build.yaml:27]: We could use the `--to-version-policy` and that would only build the published packages however we still have things like "example-code" and some other that contribute to the docs site as well.

We could probably do something better though here.

Comment (calebmshafer) [reply common/config/azure-pipelines/templates/docs-build.yaml:27]: Something like `rush build --to-version-policy prerelease-monorepo-lockStep && rush build --to example-code-snippets && rush build --to example-code-app`.

Although that being said I do want to clean up our usage of example code.

Comment (williamkbentley) [reply common/config/azure-pipelines/templates/docs-build.yaml:27]: Oh, nice. So the Rush team already thought of this use case. I'd be in favor of your suggestion. Every half a minute counts.

Comment (calebmshafer) [reply common/config/azure-pipelines/templates/docs-build.yaml:27]: Cool, let me test out that change.

Comment (williamkbentley) [reply common/config/azure-pipelines/templates/docs-build.yaml:30]: Same for `rush docs`. I'll push that.

Comment (calebmshafer) [reply common/config/azure-pipelines/templates/docs-build.yaml]: @naveedkhan8067 did you create a new pipeline to run this build? I don't see it in the previous section.

Comment (williamkbentley) [reply common/config/azure-pipelines/templates/docs-build.yaml:30]: Nevermind. Rush docs isn't as smart as rush build, and most of those "extra" packages don't have a docs script anyway, whoops.

Comment (williamkbentley) [reply common/config/azure-pipelines/templates/docs-build.yaml:30]: Saved 2.5 mins from the previous `rush compile`

**Comment by calebmshafer** [comment]
this has been superseded by PRs that have already been merged to `master`.

**Pull request closed without merging**